modules/core/src/main/java/org/apache/ignite/internal/marshaller/optimized/OptimizedObjectOutputStream.java

```diff
@@ -180,7 +181,7 @@ private void writeObject0(Object obj) throws IOException {
         if (obj == null)
             writeByte(NULL);
         else {
-            if (obj instanceof Throwable && !(obj instanceof Externalizable)) {
+            if (obj instanceof Throwable && !(obj instanceof Externalizable) || BinaryUtils.isEnum(obj.getClass())) {
```

Comment:
1. There's git annotate :)
2. Yes I don't like it either, I'll think about it, but I'd rather not create a new class for just 1 method. I'll see where I can stick it.
3. Yes since we mark this record as JDK (see below) so remote always knows how to de-serialize.